refactor: UserDetails Principal, AuthWebtyUserProvider Implement

sang218 opened the pull request:
### 요약
- `WebtyUser` 책임 분리:
    인증에서는 `webtyUserDetails`, 일반 서비스에서는 `WebtyUser`를 사용하도록 수정함
- `AuthWebtyUserProvider` 구현:
    `webtyUserDetails`로부터 `WebtyUser`를 안전하게 가져오는 클래스

### 상세
- JwtManager 수정:
    - getAuthentication 수정: webtyUserDetails 사용
- AuthWebtyUserProvider 구현:
    - `webtyUserDetails`로부터 `WebtyUser`를 안전하게 가져와서 리턴
    - 예외처리 개선 필요
- WebtyUser 수정: UserDetails 구현체 -> 도메인 User

- UserController 수정:
    - 모든 요청 처리 webtyUserDetails 사용
- UserService 수정: 
    - `AuthWebtyUserProvider` 사용
    - 사용하지 않는 주석 제거
- FavoriteController 수정:
    - `webtyUserDetails` 사용
- FavoriteService 수정: 
    - `AuthWebtyUserProvider` 사용

- WebtyUserDetailsService 수정:
    - `userRepository`를 직접 호출하지 않고 `userService` 호출로 처리
    
- UserRepository 수정: 단순 줄바꿈 수정
- UserDataResponse 수정: 주석 추가
- LoginSuccessHandler 수정: 잘못된 주석 수정


### 테스트
- 1차 수정) 기능 테스트 진행 목록:
  - [x] login
  - [x] logout
  - [x] user/info
  - [ ] delete
- 2차 수정) 실행 되는지만 확인함

### 참고
[velog에 작성한 Principal 관련 글](https://velog.io/@dia218/Principal)

---

close #68

Commit:
refactor: UserDetails Principal

### 요약
WebtyUser 책임 분리:
 인증에서는 `webtyUserDetails`, 일반 서비스에서는 `WebtyUser`를 사용하도록 수정함

### 상세
- JwtManager 수정:
    - getAuthentication 수정: webtyUserDetails 사용
- UserController 수정:
    - 모든 요청 처리 webtyUserDetails 사용
- WebtyUser 수정: UserDetails 구현체 -> 도메인 User

- UserDataResponse 수정: 주석 추가
- LoginSuccessHandler 수정: 잘못된 주석 수정
- UserService 수정: 사용하지 않는 주석 제거

### 테스트
- 기능 테스트 진행 목록:
  - [x] login
  - [x] logout
  - [x] user/info
  - [ ] delete

### 참고
[velog에 작성한 Principal 관련 글](https://velog.io/@dia218/Principal)

---

close #68

DayDreaam commented on src/main/java/org/team14/webty/user/controller/UserController.java:
```diff
@@ -27,36 +27,38 @@
-		@AuthenticationPrincipal WebtyUser webtyUser,
+		@AuthenticationPrincipal WebtyUserDetails webtyUserDetails,
 		@RequestBody @Valid NicknameRequest request) {
-		String nickname = request.getNickname();
-		userService.modifyNickname(webtyUser, nickname);
+		// if (webtyUserDetails == null) {
+		// 	return ResponseEntity.status(HttpStatus.UNAUTHORIZED)
+		// 		.body(new NicknameResponse("로그인이 필요합니다."));
+		// } //dto 수정 작업과 함께 예외 처리 추가 필요
+		userService.modifyNickname(webtyUserDetails.getWebtyUser(), request.getNickname());
```
아 @AuthenticationPrincipal을 쓰는건 맞는데 Details를 받아오고 그 다음에 진짜 WebtyUser로 바꾸는 과정이 필요하다는 거군요? 맞을까요??

sang218 replied:
네네 맞습니다!

---

예외처리도 추가하려다가 
예외처리는 dto랑 같이 일괄적으로 개선하면 좋을 것 같아서 추가하지 않고 냅뒀습니다

예외처리 클래스를 비슷한 상황에서 다르게 쓰는 등, 
예외처리 방식이 코드마다 조금씩 다른 것 같아서 이걸 어떻게 개선하면 좋을지 고민하고 있습니다

DayDreaam reviewed (commented):
확인했습니다

Dia218 commented:
관심 웹툰 부분 같이 수정해야합니다

Commit:
refactor: Implement AuthWebtyUserProvider

### 요약
`AuthWebtyUserProvider`를 이용해 `webtyUserDetails`로부터 `WebtyUser`를 안전하게 가져옴

### 상세
- AuthWebtyUserProvider 구현:
    - `webtyUserDetails`로부터 `WebtyUser`를 안전하게 가져와서 리턴
    - 예외처리 개선 필요

- FavoriteController 수정:
    - `webtyUserDetails` 사용
- FavoriteService 수정:
    - `AuthWebtyUserProvider` 사용
- UserController 수정:
    - `webtyUserDetails`를 직접 처리하지 않고 `UserService`에게 넘김 (책임 분리)
- UserService 수정:
    - `AuthWebtyUserProvider` 사용

- WebtyUserDetailsService 수정:
    - `userRepository`를 직접 호출하지 않고 `userService` 호출로 처리

- UserRepository 수정:
    - 단순 코드 최적화

### 테스트
- 실행이 되는지만 테스트 함

---

close #68

Dia218 commented:
@DayDreaam @segye 
병합 전 검토 부탁드립니다

segye reviewed (approved):
확인했습니다!

DayDreaam reviewed (approved):
확인했습니다 고생 많으셨습니다!
